c7n/resources/ssm.py

```diff
@@ -597,3 +598,54 @@ def register_resource(cls, registry, resource_class):
 
 
 resources.subscribe(PostItem.register_resource)
+
+
+@resources.register('ssm-data-sync')
+class SSMDataSync(QueryResourceManager):
+    """Resource for AWS DataSync
+    https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-inventory-datasync.html
+    """
+    class resource_type(TypeInfo):
+
+        enum_spec = ('list_resource_data_sync', 'ResourceDataSyncItems', None)
+        service = 'ssm'
+        arn_type = 'datasync'
+        id = 'DataSync'
+        name = 'Title'
+
+    permissions = ('ssm:ListResourceDataSync',)
+
+
+@SSMDataSync.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+    RelatedIdsExpression = 'S3Destination.AWSKMSKeyARN'
+
+
+@SSMDataSync.action_registry.register('delete')
+class DeleteDataSync(Action):
+    """Delete SSM data sync resources.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: delete-resource-data-sync
+            resource: ssm-data-sync
+            actions:
+              - type: delete
+    """
+
+    schema = type_schema(
+        'delete',
+        force={'type': 'boolean'}
+    )
+    permissions = ('ssm:DeleteResourceDataSync',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ssm')
+        for r in resources:
+            try:
+                client.delete_resource_data_sync(SyncName=r['SyncName'])
+            except client.exceptions.ResourceDataSyncNotFoundException as e:
+                raise(e)
```

Comment:
actually we want to continue in this case, we're trying to delete something thats already gone, we can safely continue to the next.